src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -868,7 +868,42 @@ void MFEMSidreDataCollection::Load(const std::string& path,
     std::string suffixedPath = endsWith(path, ".root") ? path : path + ".root";
 
     IOManager reader(m_comm);
-    reader.read(m_bp_grp->getDataStore()->getRoot(), suffixedPath);
+    // Get the paths in which the global and domain groups are stored
+    // This allows for a datastire structure other than the one used when this
+    // class owns the DataStore
+    Group* global_grp = m_bp_index_grp->getParent()->getParent();
+    Group* domain_grp = m_bp_grp->getParent();
+    const std::string global_path = global_grp->getPath();
+    const std::string domain_path = domain_grp->getPath();
+
+    // Create a temporary group to read the data into
+    // This is done instead of creating a temp DataStore so the buffers are intact
+    Group* temp_root =
+      m_bp_grp->getDataStore()->getRoot()->createGroup("_sidre_tmp_load");
+    reader.read(temp_root, suffixedPath);
+
+    // First transfer the global group from the temp group to its correct location
+    Group* datastore_root = m_bp_grp->getDataStore()->getRoot();
+    Group* new_global_group = temp_root->getGroup(global_grp->getPathName());
+    Group* global_group_parent = datastore_root;
+    if(!global_path.empty())
+    {
+      global_group_parent = datastore_root->getGroup(global_path);
+    }
+    // The group may already exist, but we're going to overwrite it
+    global_group_parent->destroyGroup(new_global_group->getName());
+    global_group_parent->moveGroup(new_global_group);
+    // Then transfer the domain group in the same way
+    Group* new_domain_group = temp_root->getGroup(domain_grp->getPathName());
+    Group* domain_group_parent = datastore_root;
+    if(!domain_path.empty())
+    {
+      domain_group_parent = datastore_root->getGroup(domain_path);
+    }
+    domain_group_parent->destroyGroup(new_domain_group->getName());
+    domain_group_parent->moveGroup(new_domain_group);
+    // Finally we can delete the temporary group
+    temp_root->getParent()->destroyGroup("_sidre_tmp_load");
```

Comment:
Do we need to also destroy the data/buffers under this tmp group?

Note: This might depend on https://github.com/LLNL/axom/issues/674

Reply:
This is actually intentionally omitted - the intent at least is that the data/buffers under the temporary group were transferred (via `moveGroup`) to the global and domain groups.

Reply:
In that case, please consider adding a comment,
e.g. something like: "Now that the data has been transferred, we can delete the temporary group" ?